opts/opts.go

```diff
@@ -197,20 +197,36 @@ func ValidateLink(val string) (string, error) {
 //    [host-dir:]container-path[:rw|ro]  - but doesn't validate the mode part
 func ValidatePath(val string) (string, error) {
 	var containerPath string
+	var mode string
 
 	if strings.Count(val, ":") > 2 {
 		return val, fmt.Errorf("bad format for volumes: %s", val)
 	}
 
-	splited := strings.SplitN(val, ":", 2)
-	if len(splited) == 1 {
+	splited := strings.SplitN(val, ":", 3)
+	if splited[0] == "" {
+		return val, fmt.Errorf("bad format for volumes: %s", val)
+	}
+	switch len(splited) {
+	case 1:
 		containerPath = splited[0]
-		val = path.Clean(splited[0])
-	} else {
+		val = path.Clean(containerPath)
+	case 2:
+		if splited[1] == "rw" || splited[1] == "ro" {
```

Comment:
I think we may need to externalize and then use validMountMode() from daemon/volumes.go since the list of modes is more than just rw and ro.  What do you think?